### PR TITLE
[FLINK-21095][ci] Remove legacy slot management profile

### PR DESCRIPTION
**Pull request by zentol**

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/configuration/ClusterOptions.java [hunk 1]
```diff
@@ -132,11 +132,7 @@
                             "Defines whether the cluster uses fine-grained resource management.");
 
     public static boolean isDeclarativeResourceManagementEnabled(Configuration configuration) {
```
Comment (zentol) [inline flink-core/src/main/java/org/apache/flink/configuration/ClusterOptions.java:134]: I thought about going one step further and removing this method and the `ENABLE_DECLARATIVE_RESOURCE_MANAGEMENT` option, but `ENABLE_FINE_GRAINED_RESOURCE_MANAGEMENT` makes this quite annoying so I left it as is.

Comment (tillrohrmann) [reply flink-core/src/main/java/org/apache/flink/configuration/ClusterOptions.java:134]: I don't fully understand why it is not possible. I thought that the declarative resource management is now always enabled. Hence `isFineGrainedResourceManagementEnabled` could be changed by removing the first if check which checks if declarative resource management is not activated. Then we should be able to remove the `ENABLE_DECLARATIVE_RESOURCE_MANAGEMENT` config option.

**Review by tillrohrmann — CHANGES_REQUESTED** [review]
Thanks for creating this PR @zentol. The changes look good to me. I am wondering why we can't remove the `ENABLE_DECLARATIVE_RESOURCE_MANAGEMENT` option. Just skimming over the code, it looks as if it should be possible.

**Review by tillrohrmann — APPROVED** [review]
Nice :-) +1 for merging after AZP gives green light.

**Pull request merged**